api/api.yaml

```diff
@@ -191,6 +191,26 @@ paths:
           $ref: '../result/request.yaml#/components/responses/BadRequest'
         '500':
           $ref: '../result/request.yaml#/components/responses/InternalServerError'
+  
+  /control/actions:
```

Comment:
another related question, will "flap" a link or a protocol session be considered an "action"?

Reply:
(a) Similar triggers/actions are possible for BGP, ISIS, and more protocols in the future. Note - RouteWithdraw is one such action that is already added as a separate API. For example, in BGP peers only, we could have break/resume TCP sessions, Pause/Resume Keep alive, and do graceful restart are some example. Other protocols can have similar events at different object hierarchies.
(b) Another event for LACP, Member port up/down. Probably we will add this event after more discussion.

Reply:
Currently, we have no direct way to bring down (flap) LACP link. This proposed new event would bring down the LACP link.

Reply:
@ajbalogh are we good with this action node given by Prasenjit?  what do you recommend with actions that have already gotten through via other api endpoints (Prasenjit brought up the routewithdraw example)

Reply:
To limit the spread of the api surface this action api should consolidate other single use protocol based apis  such as `set_protocol_state`, `set_route_state`, `send_ping` under one new api `set_protocol_states`. The new api should present a choice at the top level with choices for `protocol_state`, `route_state`, `lacp_pdu_state`. The choice allows for expansion of setting protocol states without increasing the top level api surface. The individual use protocol apis should be deprecated with the description for each pointing users to the new consolidated `set_protocol_states` api.

Reply:
Agree. Do the consolidation and deprecation in a different PR, in a subsequent sprint, This will help to start work on LACP implementation immediately and have the time to gradually deprecate/migrate to consolidate API.

Reply:
@PrasenjitAdhikary  the issue linked to this consolidation/deprecation should be in the next sprint. we need to clean this up before wider use.